bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/ContainerServlet.java

```diff
@@ -106,7 +109,12 @@ protected void doPost(@NotNull SlingHttpServletRequest request, @NotNull final S
     protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) throws ServletException, IOException {
         Resource container = request.getResource();
         RequestDispatcherOptions options = new RequestDispatcherOptions();
-        options.setForceResourceType(Editor.RESOURCE_TYPE);
+        String resourceType = request.getParameter(SlingConstants.PROPERTY_RESOURCE_TYPE);
```

Comment:
Should this parameter maybe be limited to a fixed set of allowed resource types? Not sure if this could be misused in some way.

For example we could make the servlet configurable as a factory, and configure the editor resource type together with the `sling.servlet.resourceTypes`. So one servlet instance would handle the requests of the wcm container with the wcm editor, another servlet instance handles the segments component of the email components with the segment editor.

Reply:
@Buuhuu IMHO this is not required. The `ContainerServlet` is already limited to the `panelcontainer` resource type (see https://github.com/adobe/aem-core-wcm-components/blob/main/bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/ContainerServlet.java#L50), so I wonder how this introduced parameter can be misused when the `ContainerServlet` can only include a panelcontainer in the get method with various resource types?